outlines/outline.py

```diff
@@ -0,0 +1,52 @@
+import ast
+from dataclasses import dataclass
+
+
+@dataclass
+class Outline:
+    """
+    Outline is a class that creates a callable object to generate responses
+    based on a given model and a prompt template.
+
+    Args:
+        model: The model to be used for generating responses.
+        template (function): A function that takes arguments and returns a prompt string.
+        output_type: The expected output type of the generated response.
+
+    Example:
+        from outlines import models
+
+        model = models.transformers("gpt2")
+
+        def template(a: int) -> str:
+            return f"What is 2 times {a}?"
+
+        fn = Outline(model, template, int)
+
+        result = fn(3)
+        print(result)  # Expected output: 6
+    """
+
+    def __init__(self, model, template, output_type):
+        self.model = model
+        self.template = template
+        self.output_type = output_type
+
+    def __call__(self, *args):
+        # Generate the prompt using the template function
+        prompt = self.template(*args)
```

Comment:
How should we support the corner case where users pass a string?

Reply:
How is that a corner case? I'll add a test to demonstrate that it already works.

Reply:
My bad, I should have reviewed more carefully